python/ray/serve/_private/application_state.py

```diff
@@ -1034,17 +1038,25 @@ def override_deployment_info(
     for options in deployment_override_options:
         deployment_name = options["name"]
         info = deployment_infos[deployment_name]
+        original_options = info.deployment_config.dict()
+        original_options["user_configured_option_names"].update(set(options))
 
-        if (
-            info.deployment_config.autoscaling_config is not None
-            and info.deployment_config.max_concurrent_queries
-            < info.deployment_config.autoscaling_config.target_num_ongoing_requests_per_replica  # noqa: E501
-        ):
-            logger.warning(
-                "Autoscaling will never happen, "
-                "because 'max_concurrent_queries' is less than "
-                "'target_num_ongoing_requests_per_replica' now."
+        if options.get("num_replicas") == "auto":
+            options["num_replicas"] = None
```

Comment:
Hmm, will this make it show up as `None` if you fetch the config from the REST API endpoint?

Reply:
Nope, `num_replicas` won't be included from the returned dict from REST API because it goes through [this code](https://github.com/ray-project/ray/blob/master/python/ray/serve/schema.py#L464-L467). I've also added this as an assert to the `num_replicas="auto"` tests.